File: 0069-sqrtx/0069-sqrtx.py
import math
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0:return 0
        guess = x
        while abs(guess * guess - x) > 0.0001:
            guess = (guess + x / guess) / 2

        return math.floor(guess)

File: 0069-sqrtx/0069-sqrtx.cpp
#include <cmath>
class Solution {
public:
    int mySqrt(int x) {
        if (x == 0) {
            return 0;
        } 
        
        double guess = x;

        while (fabs(guess * guess - x) > 1e-4) {
            guess = (guess + x / guess) / 2;
        }
        return std::floor(guess);
    }
};